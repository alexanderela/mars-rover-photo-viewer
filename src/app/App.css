#root {
  max-width: 1280px;
  margin: 0;
  text-align: center;
  width: 100%;
}

body {
  margin: 0;
}
