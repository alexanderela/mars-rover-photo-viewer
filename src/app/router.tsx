import {
  createBrowserRouter,
  redirect,
} from "react-router-dom";
import { EnlargedPhotoViewer } from "../features/EnlargedPhotoViewer";

export const routesConfig = [
  {
    path: "/",
    lazy: async () => {
      const { default: App } = await import("./App");

      return {
        element: (
          <App />
        ),
      }
    },
    children: [
      {
        index: true,
        loader: async () => redirect("rovers/curiosity?page=1")
      },
      {
        path: "rovers/:rover",
        lazy: async () => {
          const { PhotoViewer } = await import("../features/photos/PhotoViewer");
    
          return {
            element: (
              <PhotoViewer />
            )
          }
        },
        children: [
          {
            path: "id/:id",
            element: <EnlargedPhotoViewer />
          }
        ]
      },
    ]
  },
];

export const router = createBrowserRouter(routesConfig);