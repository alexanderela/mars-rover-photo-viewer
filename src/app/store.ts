import { configureStore } from "@reduxjs/toolkit";
import roverPhotosReducer from "../reducers/roverPhotosSlice";
import pageSliceReducer from "../reducers/pageSlice";
import selectedRoverSliceReducer from "../reducers/selectedRoverSlice";

const reducer = {
  roverPhotos: roverPhotosReducer,
  page: pageSliceReducer,
  selectedRover: selectedRoverSliceReducer,
}

export const store = configureStore({ 
  reducer,
});

export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];