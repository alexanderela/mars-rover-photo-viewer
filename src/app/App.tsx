import Grid from "@mui/material/Grid2";
import { Header } from "../layout/Header";
import { SideNav } from "../layout/SideNav";
import { Content } from "../layout/Content";
import "./App.css";

const App = () => {
  return (
    <>
      <Grid
        container
        sx={{
          backgroundColor: (theme) => theme.palette.primary.main,
          height: "100%",
          width: "100vw",
          padding: 0,
        }}
      >
        <Header />
        <SideNav />
        <Content />
      </Grid>
    </>
  );
};

export default App;
