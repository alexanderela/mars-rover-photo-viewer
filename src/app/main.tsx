import { createRoot } from "react-dom/client";
import { Provider } from "react-redux";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../styles/theme.ts";
import { setupStore } from "./store.ts";
import { RouterProvider } from "react-router-dom";
import { router } from "./router.tsx";
import "../index.css";
import "../reset.css";

createRoot(document.getElementById("root")!).render(
  <Provider store={setupStore()}>
    <ThemeProvider theme={theme}>
      <RouterProvider router={router} />
    </ThemeProvider>
  </Provider>,
);
