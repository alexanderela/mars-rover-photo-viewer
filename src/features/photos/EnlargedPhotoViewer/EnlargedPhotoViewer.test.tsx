import { describe, expect, it } from "vitest";
import { render,  waitFor } from "../../../tests/test-utils";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { routesConfig } from "../../../app/router";
import { mockPreloadedState } from "../../../tests/mocks";

describe("<EnlargedPhotoViewer />", () => {
  it.skip("renders EnlargedPhotoViewer component", async () => {
    const router = createMemoryRouter(routesConfig, {
      initialEntries: [ "/rovers/curiosity/id/1228212?page=1"],
    });
    
    const result = render(<RouterProvider router={router} />, {
      preloadedState: mockPreloadedState
    });

    await waitFor(() => {
      console.log("window.location.pathname: ", window.location.pathname)
      const element = result.getByTestId(/EnlargedPhotoViewer/i);
      expect(element).toBeInTheDocument();
      expect(element).toMatchSnapshot();
    });
  });
});