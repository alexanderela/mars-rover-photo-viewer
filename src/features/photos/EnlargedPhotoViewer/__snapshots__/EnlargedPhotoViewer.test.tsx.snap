// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<EnlargedPhotoViewer /> > renders EnlargedPhotoViewer component 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-1tlwrvt-MuiModal-root-MuiDialog-root"
  data-testid="EnlargedPhotoViewer"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r1:"
      aria-modal="true"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXl css-pdteti-MuiPaper-root-MuiDialog-paper"
      role="dialog"
      style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-1k2y3r"
      />
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
